### PR TITLE
zh-TW: cr2 Bare Metal translations

### PR DESCRIPTION
**Pull request by henrif75**
Professional translations for CR2 content.
Changes are being staged in a separate branch until all zh-TW content is reviewed.

#1463 #684

### CODE REVIEW

File: po/zh-TW.po [hunk 1]
```diff
@@ -16044,91 +16092,105 @@ msgid ""
 "way without properly configuring it first. Illegal state transitions are "
 "caught at compile time."
 msgstr ""
+"值的型別會指出其所處狀態，例如本例中 GPIO 接腳的設定狀態。這可將狀態機器編碼"
+"至型別系統，確保您在未事先適當設定接腳時，不會嘗試以特定方式使用接腳。在編譯"
```
Comment (johnathan79717) [inline po/zh-TW.po:16096]:
```suggestion
"至型別系統，確保您不會在未事先適當設定時嘗試使用接腳。在編譯"
```

File: po/zh-TW.po [hunk 2]
```diff
@@ -16044,91 +16092,105 @@ msgid ""
...
-msgstr ""
+msgstr "許多 HAL Crate 都遵循這個模式。"
 
 #: src/bare-metal/microcontrollers/embedded-hal.md
 msgid ""
 "The [`embedded-hal`](https://crates.io/crates/embedded-hal) crate provides a "
 "number of traits covering common microcontroller peripherals."
 msgstr ""
+"[`embedded-hal`](https://crates.io/crates/embedded-hal) Crate 提供多個特徵，"
+"可涵蓋常見的微控制器周邊裝置。"
```
Comment (johnathan79717) [inline po/zh-TW.po:16116]:
```suggestion
"涵蓋常見的微控制器周邊裝置。"
```

File: po/zh-TW.po [hunk 3]
```diff
@@ -16044,91 +16092,105 @@ msgid ""
...
 
 #: src/bare-metal/microcontrollers/embedded-hal.md
 msgid "Timers"
-msgstr ""
+msgstr "計時器"
 
 #: src/bare-metal/microcontrollers/embedded-hal.md
 msgid "Watchdogs"
-msgstr ""
+msgstr "監控計時器"
```
Comment (johnathan79717) [inline po/zh-TW.po:16140]:
```suggestion
msgstr "看門狗計時器"
```

File: po/zh-TW.po [hunk 4]
```diff
@@ -16044,91 +16092,105 @@ msgid ""
...
 "platforms such as Linux on Raspberry Pi."
 msgstr ""
+"許多微控制器和其他平台 (例如 Raspberry Pi 上的 Linux) 都有相應的實作項目。"
 
 #: src/bare-metal/microcontrollers/embedded-hal.md
 msgid ""
 "There is work in progress on an `async` version of `embedded-hal`, but it "
 "isn't stable yet."
-msgstr ""
+msgstr "`embedded-hal` 的 `async` 版本仍在開發中，但尚未推出穩定版。"
```
Comment (johnathan79717) [inline po/zh-TW.po:16163]:
```suggestion
msgstr "`embedded-hal` 的 `async` 版本已在開發中，但尚未推出穩定版。"
```

File: po/zh-TW.po [hunk 5]
```diff
@@ -16044,91 +16092,105 @@ msgid ""
...
+msgstr "Cargo 整合"
 
 #: src/bare-metal/microcontrollers/probe-rs.md
 msgid ""
 "`cargo-embed` is a cargo subcommand to build and flash binaries, log RTT "
 "(Real Time Transfers) output and connect GDB. It's configured by an `Embed."
 "toml` file in your project directory."
 msgstr ""
+"`cargo-embed` 是 Cargo 子指令，用於建構和刷新二進位檔、記錄 RTT (即時傳輸) 機"
+"制輸出內容，以及連結 GDB。這個指令是由專案目錄中的 `Embed.toml` 檔案來設定。"
```
Comment (johnathan79717) [inline po/zh-TW.po:16193]:
```suggestion
"`cargo-embed` 是 Cargo 子指令，用於建構和刷新二進位檔、記錄 RTT (即時傳輸)"
"輸出內容，以及連結 GDB。這個指令是由專案目錄中的 `Embed.toml` 檔案來設定。"
```

File: po/zh-TW.po [hunk 6]
```diff
@@ -16137,152 +16199,164 @@ msgid ""
...
+"[CMSIS-DAP](https://arm-software.github.io/CMSIS_5/DAP/html/index.html) 是針"
+"對 USB 的 Arm 標準通訊協定，供電路內偵錯工具存取各種 Arm Cortex 處理器的 "
+"CoreSight 偵錯存取埠。這就是 BBC micro:bit 內建偵錯工具所使用的通訊協定。"
 
 #: src/bare-metal/microcontrollers/probe-rs.md
 msgid ""
 "ST-Link is a range of in-circuit debuggers from ST Microelectronics, J-Link "
 "is a range from SEGGER."
 msgstr ""
+"ST-Link 是 STMicroelectronics 推出的一系列電路內偵錯工具，J-Link 則來自 "
```
Comment (johnathan79717) [inline po/zh-TW.po:16211]:
```suggestion
"ST-Link 是 ST Microelectronics 推出的一系列電路內偵錯工具，J-Link 系列則來自 "
```

File: po/zh-TW.po [hunk 7]
```diff
@@ -16137,152 +16199,164 @@ msgid ""
...
 msgid ""
 "RTT (Real Time Transfers) is a mechanism to transfer data between the debug "
 "host and the target through a number of ringbuffers."
 msgstr ""
+"RTT (即時傳輸) 這種機制是透過多個環形緩衝區，在偵錯主機和目標之間傳輸資料。"
 
 #: src/bare-metal/microcontrollers/debugging.md
 msgid "_Embed.toml_:"
-msgstr ""
+msgstr "__Embed.toml："
```
Comment (johnathan79717) [inline po/zh-TW.po:16248]:
```suggestion
msgstr "Embed.toml："
```

File: po/zh-TW.po [hunk 8]
```diff
@@ -16137,152 +16199,164 @@ msgid ""
...
 msgid "[Hubris](https://hubris.oxide.computer/)"
-msgstr ""
+msgstr "[Hubris](https://hubris.oxide.computer/)"
 
 #: src/bare-metal/microcontrollers/other-projects.md
 msgid ""
 "Microkernel RTOS from Oxide Computer Company with memory protection, "
 "unprivileged drivers, IPC"
 msgstr ""
+"Oxide Computer Company 的微核心 RTOS，提供記憶體防護、未授權的驅動程式、IPC"
```
Comment (johnathan79717) [inline po/zh-TW.po:16311]:
```suggestion
"Oxide Computer Company 的微核心 RTOS，提供記憶體防護、未具有特權的驅動程式、IPC"
```

File: po/zh-TW.po [hunk 9]
```diff
@@ -16296,10 +16370,12 @@ msgid ""
 "serial port. If you have time, try displaying it on the LEDs somehow too, or "
 "use the buttons somehow."
 msgstr ""
+"我們將讀取 I2C 羅盤中的方向，並將讀數記錄到序列埠。如有時間，可以試著顯示在 "
```
Comment (johnathan79717) [inline po/zh-TW.po:16373]:
```suggestion
"我們將讀取 I2C 羅盤上的方向，並將讀數記錄到序列埠。如有時間，可以試著顯示在 "
```

File: po/zh-TW.po [hunk 10]
```diff
@@ -16296,10 +16370,12 @@ msgid ""
...
+"LED 上，或以某種方法使用按鈕。"
 
 #: src/exercises/bare-metal/compass.md
 msgid "Hints:"
-msgstr ""
+msgstr "暗示："
```
Comment (johnathan79717) [inline po/zh-TW.po:16378]:
```suggestion
msgstr "提示："
```

File: po/zh-TW.po [hunk 11]
```diff
@@ -16308,72 +16384,83 @@ msgid ""
...
 
 #: src/exercises/bare-metal/compass.md
 msgid ""
 "The LSM303AGR driver needs something implementing the `embedded_hal::"
 "blocking::i2c::WriteRead` trait. The [`microbit::hal::Twim`](https://docs.rs/"
 "microbit-v2/latest/microbit/hal/struct.Twim.html) struct implements this."
 msgstr ""
+"LSM303AGR 驅動程式需要某個實作 `embedded_hal::blocking::i2c::WriteRead` 特徵"
+"的項目。[`microbit::hal::Twim`](https://docs.rs/microbit-v2/latest/microbit/"
+"hal/struct.Twim.html) 結構體可完成此實作。"
```
Comment (johnathan79717) [inline po/zh-TW.po:16409]:
```suggestion
"hal/struct.Twim.html) 結構體實作此特徵。"
```

File: po/zh-TW.po [hunk 12]
```diff
@@ -16308,72 +16384,83 @@ msgid ""
...
 #: src/exercises/bare-metal/compass.md src/exercises/bare-metal/rtc.md
-#, fuzzy
 msgid "_src/main.rs_:"
-msgstr "`main.rs`："
+msgstr "_src/main.rs_:"
 
 #: src/exercises/bare-metal/compass.md src/exercises/bare-metal/rtc.md
 msgid "_Cargo.toml_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__Cargo.toml (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:16442]:
```suggestion
msgstr "Cargo.toml (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 13]
```diff
@@ -16308,72 +16384,83 @@ msgid ""
...
 
 #: src/exercises/bare-metal/compass.md
 msgid "_Embed.toml_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__Embed.toml (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:16446]:
```suggestion
msgstr "Embed.toml (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 14]
```diff
@@ -16308,72 +16384,83 @@ msgid ""
...
 
 #: src/exercises/bare-metal/compass.md src/exercises/bare-metal/rtc.md
 msgid "_.cargo/config.toml_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__.cargo/config.toml (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:16450]:
```suggestion
msgstr ".cargo/config.toml (您應該不需要變更此項目)："
```

**Review by johnathan79717 — CHANGES_REQUESTED** [review]
Reviewed the morning part.

**Comment by henrif75** [comment]
> Reviewed the morning part.

Thank you very much @johnathan79717 !

File: po/zh-TW.po [hunk 15]
```diff
@@ -17291,93 +17580,112 @@ msgid ""
...
 "hypervisor), but isn't a good pattern in general."
 msgstr ""
+"如果是在 VM 中執行，可能會導致快取一致性問題。問題在於 VM 會在快取已停用時直"
+"接存取記憶體，而主機具有相同記憶體的可快取別名。即使主機未明確存取記憶體，推"
+"測存取行為仍可能導致快取填補，而存取之間的變更就會遺失。同樣地，這在本例中不"
+"成問題 (直接在硬體上執行，沒有管理程序)，但一般不建議採用這種模式。"
 
 #: src/bare-metal/useful-crates.md
+#, fuzzy
 msgid "Useful crates"
```
Comment (johnathan79717) [inline po/zh-TW.po:17590]:
```suggestion
msgid "Useful crates"
```

File: po/zh-TW.po [hunk 16]
```diff
@@ -17291,93 +17580,112 @@ msgid ""
...
-msgstr "實用的 Crate"
+msgstr "HAL Crate"
```
Comment (johnathan79717) [inline po/zh-TW.po:17591]:
```suggestion
msgstr "實用的 Crate"
```

File: po/zh-TW.po [hunk 17]
```diff
@@ -17390,17 +17698,26 @@ msgid ""
 "allocating other address space. For example, we might want to allocate MMIO "
 "space for PCI BARs:"
 msgstr ""
+"[`buddy_system_allocator`](https://crates.io/crates/buddy_system_allocator) "
+"是實作基本夥伴系統分配器的第三方 Crate。這可用來分配其他位址空間，或供 "
+"[`LockedHeap`](https://docs.rs/buddy_system_allocator/0.9.0/"
+"buddy_system_allocator/struct.LockedHeap.html) 實作 [`GlobalAlloc`](https://"
+"doc.rust-lang.org/core/alloc/trait.GlobalAlloc.html)，以便使用標準 `alloc` "
+"Crate，如[先前](../alloc.md)所見。例如，我們可能會想針對 PCI BAR 分配 MMIO 空"
```
Comment (johnathan79717) [inline po/zh-TW.po:17706]:
```suggestion
"是實作基本夥伴系統分配器的第三方 Crate。這可供 "
"[`LockedHeap`](https://docs.rs/buddy_system_allocator/0.9.0/"
"buddy_system_allocator/struct.LockedHeap.html) 實作 [`GlobalAlloc`](https://"
"doc.rust-lang.org/core/alloc/trait.GlobalAlloc.html)，以便使用標準 `alloc` "
"Crate，如[先前](../alloc.md)所見，或用來分配其他位址空間。例如，我們可能會想針對 PCI BAR 分配 MMIO 空"
```

File: po/zh-TW.po [hunk 18]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
+"的 [`core::hint::spin_loop`](https://doc.rust-lang.org/core/hint/fn."
+"spin_loop.html))。"
 
 #: src/exercises/bare-metal/rtc.md
 msgid ""
 "_Extension if you have time:_ Enable and handle the interrupt generated by "
 "the RTC match. You can use the driver provided in the [`arm-gic`](https://"
 "docs.rs/arm-gic/) crate to configure the Arm Generic Interrupt Controller."
 msgstr ""
+"__擴充功能 (如有時間)：啟用並處理因 RTC 比對而產生的中斷情形。您可以使用 "
```
Comment (johnathan79717) [inline po/zh-TW.po:17879]:
```suggestion
"擴充功能 (如有時間)：啟用並處理因 RTC 比對而產生的中斷情形。您可以使用 "
```

File: po/zh-TW.po [hunk 19]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 msgstr ""
+"請下載[練習範本](../../comprehensive-rust-exercises.zip)，並在 `rtc` 目錄中查"
+"看下列檔案。"
 
 #: src/exercises/bare-metal/rtc.md
 msgid ""
 "_src/exceptions.rs_ (you should only need to change this for the 3rd part of "
 "the exercise):"
-msgstr ""
+msgstr "__src/exceptions.rs (您應該只需為練習的第 3 部分變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17907]:
```suggestion
msgstr "src/exceptions.rs (您應該只需為練習的第 3 部分變更此項目)："
```

File: po/zh-TW.po [hunk 20]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_src/logger.rs_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__src/logger.rs (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17911]:
```suggestion
msgstr "src/logger.rs (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 21]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_src/pl011.rs_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__src/pl011.rs (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17915]:
```suggestion
msgstr "src/pl011.rs (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 22]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_build.rs_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__build.rs (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17919]:
```suggestion
msgstr "build.rs (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 23]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_entry.S_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__entry.S (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17923]:
```suggestion
msgstr "entry.S (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 24]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_exceptions.S_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__exceptions.S (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17927]:
```suggestion
msgstr "exceptions.S (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 25]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_idmap.S_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__idmap.S (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17931]:
```suggestion
msgstr "idmap.S (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 26]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_image.ld_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__image.ld (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17935]:
```suggestion
msgstr "image.ld (您應該不需要變更此項目)："
```

File: po/zh-TW.po [hunk 27]
```diff
@@ -17502,92 +17841,106 @@ msgstr "完成練習後，您可以看看我們提供的[解決方案](solutions
...
 
 #: src/exercises/bare-metal/rtc.md
 msgid "_Makefile_ (you shouldn't need to change this):"
-msgstr ""
+msgstr "__Makefile (您應該不需要變更此項目)："
```
Comment (johnathan79717) [inline po/zh-TW.po:17939]:
```suggestion
msgstr "Makefile (您應該不需要變更此項目)："
```

**Pull request merged**